zeitschrift-fur-kunstgeschichte.csl

```diff
@@ -228,14 +253,29 @@
                 </group>
                 <names variable="editor translator" delimiter=", ">
                   <label form="verb-short" suffix=" "/>
-                  <name/>
+                  <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never"/>
                 </names>
               </if>
+              <else-if match="any" variable="translator">
+                <choose>
+                  <if match="none" variable="author">
+                    <names variable="editor">
+                      <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never"/>
+                      <label form="short" prefix=" (" suffix=")"/>
+                    </names>
+                  </if>
+                </choose>
+                <group delimiter=" ">
+                  <text variable="title" font-style="italic" suffix=","/>
+                  <text macro="translator"/>
+                  <text macro="collection-with-number"/>
+                </group>
+              </else-if>
```

Comment:
I am not sure if I understand your suggestion correctly but it would not work properly because of the structure of the macro "translator" - with edited books it would mention the editors twice. The macro "translator" is a renamed copy of "Kritische Ausgabe"'s macro "editor".